src/Sarif.Driver/OrderedFileSpecifier.cs

```diff
@@ -55,6 +55,12 @@ private IEnumerable<IEnumeratedArtifact> EnumeratedArtifacts()
             string filter = Path.GetFileName(normalizedSpecifier);
             string directory = Path.GetDirectoryName(normalizedSpecifier);
 
+            if (filter != "*" && filter != "" && !FileSystem.FileExists(normalizedSpecifier))
```

Comment:
>filter [](http://example.com/codeflow?start=33&length=6)

Under what circumstances will `Path.GetFileName` return an empty string? Did you observe one?

Reply:
Of course, makes sense! Just do a length check here against 0, `filter.Length != 0', You could perform a comparison against the `string.Empty` constant as well (but the length check is more direct). It used to be that we expressly forbade use of `""` in code as it would result in a string literal compiled into MSIL where .NET already defines one (that's interned) in `string.Empty`. In practice, all this concern around the empty string may be a bit of theater. @nguerrera may have some comments.

In generally, comparisons to the empty string are interesting because this operation might mask a need for expanded comparison semantics, for example, checks for whitespace literals. That isn't the case here as we're exclusively working against the return values of `Path.GetFileName`.